lib/utils/errors.js

```diff
@@ -61,81 +37,48 @@ const mapFunctionFailureResponseToError = ({ error, id }) => {
 };
 
 const mapApiResponseToError = ({ code, detail }) => {
-  if (code === 'unauthorized') {
-    throw new UnauthorizedError(detail);
-  }
-  if (code === 'forbidden') {
-    throw new ForbiddenError(detail);
-  }
-  if (code === 'unprocessable-content') {
-    throw new DecryptError(detail);
-  }
   if (code === 'functions/request-timeout') {
     throw new FunctionTimeoutError(detail);
   }
   if (code === 'functions/function-not-ready') {
     throw new FunctionNotReadyError(detail);
   }
-  if (code === 'functions/forbidden-ip') {
-    throw new ForbiddenIPError(detail);
-  }
-  if (code === 'decrypt/forbidden') {
-    throw new DecryptForbiddenError(detail);
```

Comment:
If I understand this correctly, with this change if the user tries to decrypt with a role that doesn't have permission they'll will receive a generic `EvervaultError` with a 403 status code that will have the message `The API key provided does not have the required permissions.`? If that's the case the message is misleading since it's not actually not an API key issue.

Reply:
They'll receive the EvervaultError with the message populated from detail, which would be populated by the API (and thus should be specific to the error the fact that the decrypt was forbidden)